File: download-destination-images.py
#!/usr/bin/env python3
"""
Download destination images from Unsplash API
"""

import os
import requests
from pathlib import Path

# Destination slugs and search queries
destinations = {
    "cape-agulhas": "Cape Agulhas lighthouse Africa ocean",
    "bo-kaap": "Bo-Kaap Cape Town colorful buildings",
    "cape-town-highlights": "Cape Town Table Mountain aerial view",
    "cape-winelands": "Cape Winelands Stellenbosch vineyards",
    "cape-peninsula": "Cape Peninsula Cape Point dramatic cliffs",
    "hermanus": "Hermanus whale watching coast South Africa",
    "aquila-safari": "Aquila game reserve lions safari Africa",
    "constantia-wine": "Constantia wine valley vineyards mountains",
    "cape-west-coast": "West Coast Langebaan lagoon beach",
    "garden-route": "Garden Route Tsitsikamma forest coast"
}

# Create destination images directory
dest_dir = Path("frontend/public/images/destinations")
dest_dir.mkdir(parents=True, exist_ok=True)

# Unsplash API endpoint (no key required for basic usage)
UNSPLASH_URL = "https://api.unsplash.com/search/photos"

def download_image(query, filename):
    """Download image from Unsplash"""
    try:
        # Search for images
        params = {
            "query": query,
            "per_page": 1,
            "orientation": "landscape"
        }
        
        response = requests.get(UNSPLASH_URL, params=params, timeout=10)
        response.raise_for_status()
        
        data = response.json()
        
        if data.get("results"):
            image_url = data["results"][0]["urls"]["regular"]
            
            # Download the image
            img_response = requests.get(image_url, timeout=10)
            img_response.raise_for_status()
            
            # Save the image
            filepath = dest_dir / filename
            with open(filepath, "wb") as f:
                f.write(img_response.content)
            
            print(f"✓ Downloaded: {filename}")
            return True
        else:
            print(f"✗ No results for: {query}")
            return False
            
    except Exception as e:
        print(f"✗ Error downloading {filename}: {str(e)}")
        return False

def main():
    """Download all destination images"""
    print("Downloading destination images from Unsplash...")
    print(f"Saving to: {dest_dir.absolute()}\n")
    
    success_count = 0
    for slug, query in destinations.items():
        filename = f"{slug}.jpg"
        if download_image(query, filename):
            success_count += 1
    
    print(f"\n✓ Successfully downloaded {success_count}/{len(destinations)} images")

if __name__ == "__main__":
    main()

File: download-destination-images.js
const https = require('https');
const fs = require('fs');
const path = require('path');

// Create destinations folder if it doesn't exist
const destDir = path.join(__dirname, 'frontend/public/images/destinations');
if (!fs.existsSync(destDir)) {
  fs.mkdirSync(destDir, { recursive: true });
}

// Map of destination slugs to image URLs from Unsplash
const destinationImages = {
  'cape-agulhas': 'https://images.unsplash.com/photo-1605662235062-31d31bd23533?w=1200&q=80', // Lighthouse
  'bo-kaap': 'https://images.unsplash.com/photo-1551019231-c4b8c9a8f8d7?w=1200&q=80', // Colorful houses
  'cape-town-highlights': 'https://images.unsplash.com/photo-1516426122078-c23e76319801?w=1200&q=80', // Table Mountain
  'cape-winelands': 'https://images.unsplash.com/photo-1510812431401-41d2cab2707d?w=1200&q=80', // Vineyards
  'cape-peninsula': 'https://images.unsplash.com/photo-1507003211169-0a1dd7228f2d?w=1200&q=80', // Coastal drive
  'hermanus': 'https://images.unsplash.com/photo-1547521064-7e2b82b339cc?w=1200&q=80', // Whale watching
  'aquila-safari': 'https://images.unsplash.com/photo-1516426122078-c23e76319801?w=1200&q=80', // Wildlife
  'constantia-wine': 'https://images.unsplash.com/photo-1510812431401-41d2cab2707d?w=1200&q=80', // Wine valley
  'cape-west-coast': 'https://images.unsplash.com/photo-1507003211169-0a1dd7228f2d?w=1200&q=80', // Coastal scenery
  'garden-route': 'https://images.unsplash.com/photo-1469022563149-aa64dbd37dae?w=1200&q=80' // Forest
};

function downloadImage(url, filename) {
  return new Promise((resolve, reject) => {
    const filepath = path.join(destDir, filename);
    
    https.get(url, (response) => {
      if (response.statusCode === 200) {
        response.pipe(fs.createWriteStream(filepath))
          .on('finish', () => {
            console.log(`✓ Downloaded ${filename}`);
            resolve();
          })
          .on('error', reject);
      } else {
        reject(new Error(`HTTP ${response.statusCode}`));
      }
    }).on('error', reject);
  });
}

async function downloadAllImages() {
  console.log('Starting destination image downloads...\n');
  
  for (const [slug, url] of Object.entries(destinationImages)) {
    const filename = `${slug}.jpg`;
    try {
      await downloadImage(url, filename);
    } catch (error) {
      console.error(`✗ Failed to download ${filename}: ${error.message}`);
    }
  }
  
  console.log('\n✓ All images downloaded!');
}

downloadAllImages().catch(console.error);
